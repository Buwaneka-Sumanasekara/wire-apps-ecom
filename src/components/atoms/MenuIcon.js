import { Ionicons } from '@expo/vector-icons';
import { Text, View } from 'react-native';

const MenuIcon = (props) => {
    const { name, size, color, value, testID } = props;

    if (value) {
        return (
            <View testID={testID} className={"relative py-2"}>
                <View className={"t-0 absolute -right-2 top-1 z-10 "}>
                    <View className={"bg-red-500 rounded-full w-4 h-4 ju"}>
                        <Text className="text-white text-center">{`${value}`}</Text>
                    </View>
                </View>
                <Ionicons name={name} size={size} color={color} />
            </View>
        );
    }

    return <Ionicons testID={testID} name={name} size={size} color={color} />;
}



export default MenuIcon;