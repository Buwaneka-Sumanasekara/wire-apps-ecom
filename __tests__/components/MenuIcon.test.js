import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react-native'

import MenuIcon from '../../src/components/atoms/MenuIcon';




describe('<MenuIcon />', () => {
    it('rendered sucessfully', () => {
        const tree = renderer.create(<MenuIcon />).toJSON()
        expect(tree).toMatchSnapshot()
    });
    it('Show number when value prop is there', () => {
        const tree = renderer.create(<MenuIcon value={10} />).toJSON()
        const testInstance = tree.children[0].children[0].children[0].children[0];
        expect(testInstance).toBe("10");
    });

});

