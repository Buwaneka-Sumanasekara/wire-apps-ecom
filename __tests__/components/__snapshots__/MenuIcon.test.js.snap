// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuIcon /> rendered sucessfully 1`] = `<Text />`;
